### ISSUE TITLE
Share build-time resources between sidecars

### ISSUE THREAD

**Issue by benoitf** [eclipse-che/che#18364]
### Is your enhancement related to a problem? Please describe.
sidecars have entrypoint for example and most of sidecars have the same single entrypoint.sh
This should be factorized to have only one source of truth allowing to update sidecars with a single change

### Describe the solution you'd like
Share resources like entrypoint.sh across several sidecars as they're in the same repository now

### Describe alternatives you've considered
N/A

### Additional context
Plugin Automation #15819

**Comment by gattytto** [comment]
the sidecars folder in plugin-registry could have a "common" folder with entrypoint.sh for them all. Then in every language inside "sidecars" there could be a etc/ folder for sidecars that have additional files in entrypoint stage.

**Comment by sleshchenko** [comment]
assigning p1 since it's what the related epic has

**Comment by ericwill** [comment]
> the sidecars folder in plugin-registry could have a "common" folder with entrypoint.sh for them all. Then in every language inside "sidecars" there could be a etc/ folder for sidecars that have additional files in entrypoint stage.
> 
> ```shell
> che-plugin-registry/sidecars/common/etc/storage.json
> che-plugin-registry/sidecars/common/etc/entrypoint.sh
> che-plugin-registry/sidecars/haskell/etc/ghcide-ghc810.yaml
> che-plugin-registry/sidecars/haskell/Dockerfile
> che-plugin-registry/sidecars/haskell/etc/entrypoint-patches.sh
> ```

Yes, the idea is to use a fragment that all container definitions can make use of.
